src/main/java/org/jboss/remoting3/EndpointImpl.java

```diff
@@ -686,14 +687,15 @@ private static <T> UnaryOperator<T> and(final UnaryOperator<T> first, final Unar
         return t -> second.apply(first.apply(t));
     }
 
-    public Registration addConnectionProvider(final String uriScheme, final ConnectionProviderFactory providerFactory, final OptionMap optionMap) throws IOException {
+    public Registration addConnectionProvider(final String uriScheme, final ConnectionProviderFactory providerFactory, final OptionMap optionMapIn) throws IOException {
         final SecurityManager sm = System.getSecurityManager();
         if (sm != null) {
             sm.checkPermission(RemotingPermission.ADD_CONNECTION_PROVIDER);
         }
         boolean ok = false;
         resourceUntick("Connection provider for " + uriScheme);
         try {
+            final OptionMap optionMap = EndpointBuilder.merge(optionMapIn, defaultConnectionOptionMap);
```

Comment:
I'm not sure changing the method parameter "optionMap" to "optionMapIn" helps in any way. It's clear from line 698 that the variable is being updated via the merge.

Reply:
The pure reason for the parameter renaming is to reduce code changes, because the method parameter is final, I would rather use a different parameter name.